tests/test_views.py

```diff
@@ -142,3 +145,88 @@ def test_success_template_change_with_gzip(self):
         event = next(response_iterable)
         assert event == b'data: {"type": "reload"}\n\n'
         assert not views.should_reload_event.is_set()
+
+
+django_4_2_plus = skipIf(django.VERSION < (4, 2), "Requires Django 4.2+")
+
+
+@override_settings(DEBUG=True)
+class AsyncEventsTests(SimpleTestCase):
+    @override_settings(DEBUG=False)
+    async def test_fail_not_debug(self):
+        response = await self.async_client.get("/__reload__/events/")
+
+        assert response.status_code == HTTPStatus.NOT_FOUND
+
+    async def test_fail_not_accepted(self):
+        response = await self.async_client.get(
+            "/__reload__/events/", ACCEPT="text/html"
+        )
+
+        assert response.status_code == HTTPStatus.NOT_ACCEPTABLE
+
+    @django_4_2_plus
+    async def test_success_ping(self):
+        response = await self.async_client.get("/__reload__/events/")
+        assert isinstance(response, StreamingHttpResponse)
+
+        assert response.status_code == HTTPStatus.OK
+        assert response["Content-Type"] == "text/event-stream"
+
+        if sys.version_info >= (3, 10):
+            event = await anext(aiter(response))
+        else:
+            event = await response.__aiter__().__anext__()
```

Comment:
repeating this pattern is a bit verbose... adding backports of `anext` / `aiter`